core/services/run_executor.go

```diff
@@ -54,10 +55,6 @@ func (je *runExecutor) Execute(runID *models.ID) error {
 			taskRun.ApplyOutput(result)
 			run.ApplyOutput(result)
 
-			if !result.Status().Runnable() {
```

Comment:
I'm not sure this is ok. I believe that the calls above can change the status again such that `Runnable()` is false.

ie: the following lines could change Status

```golang
			result := je.executeTask(&run, taskRun)
			taskRun.ApplyOutput(result)
			run.ApplyOutput(result)
```

My memory of the code could very well be outdated, so I would double check.

Reply:
It is only a `logger.Debugw` statement so not the end of the world.

Reply:
The following lines could change the status, but if you follow the linear expanded flow, the code used to look like this:

```
result := je.executeTask(&run, taskRun)
taskRun.ApplyOutput(result)
run.ApplyOutput(result)
if finished {
  logger.Debug
}
if finished {
  continue
}
```

now it is:
```
result := je.executeTask(&run, taskRun)
taskRun.ApplyOutput(result)
run.ApplyOutput(result)
if finished {
  logger.Debug
  continue
}
```